URBNCreativeTestRig/TextDetailTogglesViewController.swift

```diff
@@ -0,0 +1,359 @@
+//
+//  TextDetailTogglesViewController.swift
+//  URBNCreativeTestRig
+//
+//  Created by Jason Grandelli on 11/9/15.
+//  Copyright © 2015 Jason Grandelli. All rights reserved.
+//
+
+import UIKit
+
+class OptionTextField: UITextField {
+    var setTextBlock: () -> (Void) = {}
```

Comment:
I really like this but here’s a thought: Would it be possible to define this with an `OptionTextField` parameter (instead of `Void`). Then in your `didSet` block call `setTextBlock(self)`. That way when you implement the setTextBlock you could get at the object being set without having to reference self.